### DISCUSSION TITLE
Subgraphs autoscaling

### DISCUSSION THREAD

**Discussion by DGolubets** [apollographql/router#5698] [Q&A]
One of our subgraph services has auto-scaling configured (in K8s). 
When scaled down, Apollo Router logs errors like `fetch_error=SubrequestHttpError { status_code: None, service: "model", reason: "connection closed before message completed" }`.

Is there something I can do to handle this more gracefully? 
Maybe Apollo Router should be aware of scaling, or the subgraph service should wait longer before terminating?

**Comment by Geal** [comment]
There is no support for service discovery in the router, but you can look into supporting graceful shutdown in the subgraphs (or in the load balancer in front of them). When a subgraph is going to stop, it should return `Connection: close` on all responses for a while and stop accepting new connections. Traffic to that subgraph instance will progressively die down and be dispatched to the other instances
